### DISCUSSION TITLE
Gradle is stuck on version 7.5

### DISCUSSION THREAD

**Discussion by ronyarbiv** [airsdk/Adobe-Runtime-Support#3197] [Q&A]
Hi,

I have donloaded the last gradle, i can see in my project that the gradle is 8.7, when runing gradle -v i am getting 8.7. but in AndroidStudioProject that created after the build, the gradle is 7.5. ( i am using latest air sdk, jdk 11.0.8) in android studio(latest) i have pointed to the gradle folder. enviromental varable as well (checked with gradle -v)

Any idea? do i need to configure that in the sdk somehow? 

Thanks in advence,
Rony

**Comment by ajwfrost** [comment]
You can actually change the gradle version used by the AIR SDK build process if you add a new flag into your application descriptor XML file, within the `<android>` section, e.g.
```
<android>
  <gradleVersion>8.7</gradleVersion>
  <androidGradlePluginVersion>8.0.2</androidGradlePluginVersion>
</android>
```

I'm not 100% sure, there may be some issues if you use something that recent, we'd seen some warnings about that - I'll ask someone to check and see if there are updates we need to make for the latest versions of everything..

thanks

**Reply by ronyarbiv** [reply]
Hi,

Thank you for the suggestion. Setting the minimum SDK to 24, as recommended by @marchbold, resolved the issue running with Gradle 8.7. However, approximately 10% of my users (and thus a portion of my income) come from devices using SDK versions 21, 22, and 23. I attempted to maintain a lower minimum SDK  21 by setting the Gradle version in the XML, but then I encountered this error.

I admit that I'm not very familiar with system architecture. Is there a straightforward fix available, or would it be advisable to stick with the minimum SDK set to 24?

unexpected failure: Unable to run java: com.adobe.air.ADTException: gradle tool failed: FAILURE: Build failed with an exception. * What went wrong: A problem occurred configuring project ':androidx.activity'. > Failed to notify project evaluation listener. > Could not create an instance of type com.android.build.api.variant.impl.LibraryVariantBuilderImpl. > Namespace not specified. Please specify a namespace in the module's build.gradle file like so: android { namespace 'com.example.namespace' } If the package attribute is specified in the source AndroidManifest.xml, it can be migrated automatically to the namespace value in the build.gradle file using the AGP Upgrade Assistant; please refer to https://developer.android.com/studio/build/agp-upgrade-assistant for more information. > Could not get unknown property 'generateReleaseBuildConfig' for project ':androidx.activity' of type org.gradle.api.Project. * Try: > Run with --stacktrace option to get the stack trace. > Run with --info or --debug option to get more log output. > Run with --scan to get full insights. > Get more help at https://help.gradle.org. BUILD FAILED in 12s java.io.IOException: Unable to run java: com.adobe.air.ADTException: gradle tool failed: FAILURE: Build failed with an exception. * What went wrong: A problem occurred configuring project ':androidx.activity'. > Failed to notify project evaluation listener. > Could not create an instance of type com.android.build.api.variant.impl.LibraryVariantBuilderImpl. > Namespace not specified. Please specify a namespace in the module's build.gradle file like so: android { namespace 'com.example.namespace' } If the package attribute is specified in the source AndroidManifest.xml, it can be migrated automatically to the namespace value in the build.gradle file using the AGP Upgrade Assistant; please refer to https://developer.android.com/studio/build/agp-upgrade-assistant for more information. > Could not get unknown property 'generateReleaseBuildConfig' for project ':androidx.activity' of type org.gradle.api.Project. * Try: > Run with --stacktrace option to get the stack trace. > Run with --info or --debug option to get more log output. > Run with --scan to get full insights. > Get more help at https://help.gradle.org. BUILD FAILED in 12s at com.adobe.air.apk.AABOutputStream.buildGradle(AABOutputStream.java) at com.adobe.air.apk.AABPackager.renameOutput(AABPackager.java) at com.adobe.air.ApplicationPackager.createPackage(ApplicationPackager.java) at com.adobe.air.apk.AABPackager.createPackage(AABPackager.java) at com.adobe.air.ADT.parseArgsAndGo(ADT.java) at com.adobe.air.ADT.run(ADT.java) at com.adobe.air.ADT.main(ADT.java)

**Reply by marchbold** [reply]
Hi @ronyarbiv , 

I think if you are looking to support older versions of Android you should stick to the previous versions of the extension. As we update the extensions the underlying libraries and SDKs update their requirements, including the minimum version supported. 

I'll double check this particular instance but from memory this is the new minimum for some of the libs we are using here.

**Reply by ronyarbiv** [reply]
Thank you, 

I really appreciate your Ane they let me work freely.

 I don't know if it's alot to ask, but it will be great if MinSdk will be part of the documents of the popular ones (ads and so on) 

(maybe it is and i didn't see, so please direct me) 

Thank you again for all your great work for the community.

**Reply by marchbold** [reply]
@ronyarbiv We don't currently document it anywhere (will look into it to see if there's something we can do here), however apm will correctly insert the minimum sdk version if you use it to generate your app descriptor.

**Reply by ronyarbiv** [reply]
Thank you.

I am using the manual way. Because most of my projects are from the long past. And i always fear of problems as i don't understand much in this area. I will try and start using the apm for the new ones.

**Reply by marchbold** [reply]
Sure thing, you can have a read through this article I wrote a while ago about migrating to apm if you are interested: https://github.com/airsdk/apm/discussions/75
